pkg/restore/restore.go

```diff
@@ -1249,13 +1249,36 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		errs.Add(namespace, err)
 		return warnings, errs
 	}
-	if isAlreadyExistsError {
-		fromCluster, err := resourceClient.Get(name, metav1.GetOptions{})
-		if err != nil {
-			ctx.log.Infof("Error retrieving cluster version of %s: %v", kube.NamespaceAndName(obj), err)
-			warnings.Add(namespace, err)
+	// check if we want to treat the error as a warning, in some cases the creation call might not get executed due to object API validations
+	// and Velero might not get the already exists error type but in reality the object already exists
+	objectExists := false
+	fromCluster, err := resourceClient.Get(name, metav1.GetOptions{})
+	if err != nil {
```

Comment:
So this return without adding a new error on error should really only be followed if isAlreadyExistsError is true. Otherwise, if there's a genuine create error not related to already existing, we expect an error here too.

Reply:
I'm thinking we only want to make this Get call here if isAlreadyExistsError is false. Then we can safely ignore the error and pass through to the normal "object doesn't exist" error processing.